packages/server/migrations/1763395869093-crypto-hdindex.js

```diff
@@ -0,0 +1,52 @@
+const db = require('./db')
+const { loadConfig } = require('./settings')
+
+const GET_MAX_CRYPT_HDINDEX = `
+SELECT crypto_code, MAX(hd_index) AS hd_index
+FROM cash_out_txs
+GROUP BY crypto_code;
+`
+
+const DROP_HDIDX_SEQ = `
+DROP SEQUENCE hd_indices_seq;
+`
+
+const CREATE_HDIDX_SEQ_TABLE = `
+CREATE TABLE hd_indices_seq (
+  crypto_code TEXT NOT NULL UNIQUE,
+  hd_index INTEGER NOT NULL
+);
+`
+
+const INSERT_HDIDX = `
+INSERT INTO hd_indices_seq (crypto_code, hd_index)
+VALUES ($1, $2);
+`
+
+const insertHDIndexes = async maxHdIndex => {
+  for (const [cryptoCode, hdIndex] of Object.entries(maxHdIndex))
+    await db.none(INSERT_HDIDX, [cryptoCode, hdIndex])
+}
+
+exports.up = next =>
+  Promise.all([db.any(GET_MAX_CRYPT_HDINDEX), loadConfig()])
+    .then(([rows, { locale_cryptoCurrencies: cryptoCurrencies }]) => {
+      const maxHdIndex = rows.reduce((o, { crypto_code, hd_index }) => {
+        o[crypto_code] = hd_index ?? 0
+        return o
+      }, {})
+
+      cryptoCurrencies.forEach(cryptoCode => {
+        maxHdIndex[cryptoCode] ??= 0
+      })
+
+      return maxHdIndex
+    })
+    .then(maxHdIndex =>
+      db
+        .runAll([DROP_HDIDX_SEQ, CREATE_HDIDX_SEQ_TABLE])
+        .then(() => insertHDIndexes(maxHdIndex)),
+    )
+    .then(next)
+
```

Comment:
This can be greatly simplified by just querying this:

```sql
select crypto_code, coalesce(max(hd_index), 0)
from cash_out_txs
group by crypto_code;
```

then insert that into the new table and no js is required for this migration